master/pkg/actor/pool/pool.go

```diff
@@ -0,0 +1,168 @@
+package pool
+
+import (
+	"github.com/determined-ai/determined/master/pkg/actor"
+)
+
+// QueueFullError is returned by SubmitTask when the pool is so backed up that the queue is full.
+type QueueFullError struct{}
+
+func (e QueueFullError) Error() string {
+	return "actor pool queue is full"
+}
+
+// ActorPool implements thread-pool like behavior in the actor system. It is given a function to
+// handle tasks, and will call this function with each task submitted to the actor pool.
+type ActorPool struct {
+	name string
+
+	taskHandler func(task interface{}) interface{}
+	callback    func(result interface{})
+
+	queueLimit   uint
+	workersLimit uint
+	workers      uint
+	counter      uint64
+
+	manager *actor.Ref
+	system  *actor.System
+
+	queue chan interface{}
+}
+
+// NewActorPool initializes a new actor pool and starts its manager actor.
+func NewActorPool(
+	system *actor.System,
+	queueLimit uint,
+	workersLimit uint,
+	name string,
+	taskHandler func(task interface{}) interface{},
+	callback func(result interface{})) ActorPool {
+	pool := ActorPool{
+		name: name,
+
+		taskHandler: taskHandler, // May be called many times in parallel
+		callback:    callback,    // Will only be called by the manager actor
+
+		queueLimit:   queueLimit,
+		workersLimit: workersLimit,
+
+		system: system,
+
+		queue: make(chan interface{}, queueLimit),
+	}
+	ref, _ := system.ActorOf(actor.Addr(name), &pool)
```

Comment:
last comment: maybe make this like `/pool/{name}` or something instead?

Reply:
No can do, because /pool/ would need to be an actor (it verifies that every node in the hierarchy is an actor that exists - I personally wouldn't argue if we wanted to remove that, but for all I know right that's a good reason for it).

What I actually wanted to do originally was pass a full address, so that the HP importance actor (the one outside of the actor pool) could make it a child of itself. But the problem is it instantiates the pool when it's creating itself, and instantiating the pool requires that the parent actor already exist and be registered with IT'S address, so you get a circular dependency.

So I ended up just leaving it with the requirement that your actor pool have a globally unique name that we have now.